addons/sale_timesheet/models/project_overview.py

```diff
@@ -424,6 +424,8 @@ def _plan_get_stat_button(self):
 
         if len(task_projects_ids) == 1:
             tasks_context = {**tasks_context, 'default_project_id': task_projects_ids[0]}
+        elif not task_projects_ids and len(self) == 1:
+            tasks_context = {**tasks_context, 'default_project_id': self.id}
```

Comment:
Maybe you can check ``len(self) == 1`` first, i.e.

      if len(self.ids) == 1:
            tasks_context = {**tasks_context, 'default_project_id': self.id}
      elif len(self.ids):
        # filter out all the projects that have no tasks
        ...

Reply:
thanks for the suggestions, indeed some refactoring was possible to avoid redundant checks

Reply:
Not sure that you can remove previous logic. You should address a question to PM, if you want to change it. Or at least update commit message. 
New version doesn't work in the same way in scenario when you remove filter in "Project overview": if only one project has tasks, then before you commit default project is that project, but after your commit there is no default project

There is a missed step on the picture (between 1 and 2 steps): add a fitler that search results for severals projects and only one of those projects has tasks
![screenshot-4541009-13-0 runbot50 odoo com-2020 10 13-17_29_25](https://user-images.githubusercontent.com/186131/95882306-d4f59400-0d79-11eb-83a2-9d881c30d4dd.png)